medical_appointment/models/medical_appointment.py

```diff
@@ -0,0 +1,302 @@
+# -*- coding: utf-8 -*-
+# © 2016 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import fields, models, exceptions, api, _
+from datetime import timedelta
+
+
+class MedicalAppointment(models.Model):
+    _name = 'medical.appointment'
+    _inherit = 'medical.history.abstract'
+    _description = 'Medical Appointments'
+
+    STATES = {
+        'draft': [('readonly', False)]
+    }
+
+    user_id = fields.Many2one(
+        string='Responsible',
+        help='User that created appointment',
+        comodel_name='res.users',
+        readonly=True,
+        states=STATES,
+        default=lambda s: s.env.user,
+    )
+    patient_id = fields.Many2one(
+        string='Patient',
+        comodel_name='medical.patient',
+        required=True,
+        select=True,
+        help='Patient Name',
+    )
+    name = fields.Char(
+        string='Appointment ID',
+        default='/',
+        readonly=True,
+    )
+    force_schedule = fields.Boolean(
+        help='Check this to ignore any double bookings and schedule anyways',
+    )
+    appointment_date = fields.Datetime(
+        string='Date and Time',
```

Comment:
I suggest to make this required=True so date_end can be computed. Otherwise I was getting on error message. Also, it seems quite strange to me creating an appointment without even having an hing no the date.